### DISCUSSION TITLE
How to use filebeat and kafka for logs ?

### DISCUSSION THREAD

**Discussion by chunhaiyangzhili** [apache/skywalking#10358] [Q&A]
希望wusheng大大能给予帮助，我的英文不太好，简而言之就是我使用springboot+filebeat+kafka+skywalking+es来做日志落盘
但是依据官网提供的filebeat.yml，如图1和2，最终写到es的字段是写死的，如图3，我想加入一些字段存储到es中，比如serviceName，但是没找到kafka到es的字段映射是在哪设置的，而且我发现es中的字段，例如：tag_raw_data是base64加密的，也想不加密，或者去掉一些例如service_instance_id这种对我没有帮助的字段，但是不知道如何修改，希望能给予帮助，万分感谢！以下是翻译：

I hope wusheng can help me a lot. My English is not very good. In short, I use spring boot+filebeat+kafka+skywalking+es to do log store.
However, according to the filebeat.yml provided on the official website, as shown in Figure 1 and 2, the fields written to es are dead. As shown in Figure 3, I want to add some fields to store in es, such as serviceName, but I can't find where the field mapping from kafka to es is set, and I find the fields in es, such as tag_ raw_ The data is base64 encrypted, and you also want to not encrypt it, or remove some services, for example_ instance_ ID is a field that doesn't help me, but I don't know how to modify it. I hope you can help me. Thank you very much!

image1 and image2
![image](https://user-images.githubusercontent.com/35128699/217233220-854b2424-b7af-4b32-8578-b13f776309e2.png)
![image](https://user-images.githubusercontent.com/35128699/217233607-7e7985cd-a116-4a84-8787-232346d7465c.png)

image3
![image](https://user-images.githubusercontent.com/35128699/217233350-644c2efe-edf6-49dc-946b-60f1f1530656.png)

**Comment by wu-sheng** [comment]
A simple answer is, if you want to change codes, you are granted by Apache 2.0, but you are on your own. We don't have obligations to support that.

**Reply by chunhaiyangzhili** [reply]
So what do you mean, there is no ready-made configuration to set? Do you need to modify the source code yourself? If you want to modify the source code, can you tell me where to modify it? thank you very much indeed

**Reply by wu-sheng** [reply]
You are asking for changing, of course you need to change codes.
And as  community, we wouldn't do this kind of guidances, we can't afford.

**Reply by chunhaiyangzhili** [reply]
I see what you mean. Thank you very much. Skywalking is very good and has helped me a lot. I also want to ask where else can I find help besides the community?
